### ISSUE TITLE
how about consider migrating to `mdbook` or similar?

### ISSUE THREAD

**Issue by AndyJado** [NixOS/nix.dev#423]
I don't know about how to put it , but the website appears kinda messy, the actual readable region is awkwardly thin.

if it is just me then plz pardon me for being annoying

**Comment by yukiisbored** [comment]
I don't think this warrants changing the entire static site generator as Sphinx allows customization of how the site looks.

Currently, we're using [sphinx-book-theme] and I imagine in the future, we'll be customizing it (or making our own theme) to make it look like an official Nix website but currently the [Nix Marketing team](https://nixos.org/community/teams/marketing.html) is busy on making the main nixos.org website maintainable.

Currently, I'm working on improving the sites' ergonomics:

- https://github.com/NixOS/nix.dev/pull/409
- https://github.com/NixOS/nix.dev/pull/405

It would be incredibly helpful if you provide more input on what could be changed to make the site work for you better.

[sphinx-book-theme]: https://sphinx-book-theme.readthedocs.io/en/stable/

**Comment by willbush** [comment]
As a fan of mdbook myself, I'm pretty impressed with how nix.dev looks currently. I imagine improvements have been made since this issue was opened. IMO this should be closed since and reopened if the issue comes up again.

**Comment by fricklerhandwerk** [comment]
For reference, here's a broader investigation into aligning our documentation infrastructure:
- https://discourse.nixos.org/t/a-roadmap-for-the-documentation-ecosystem/42328
- https://ethercalc.net/dc4vcnnl8zv0

tl;dr it's not that simple

Having worked with mdBook quite a lot for the Nix reference manual, I'd rather avoid that. Sphinx is better, but not that much. At our scale and existing level of customization, rolling our own is not wrong any more, even if it sounds scary. Please [read the `nix-render-docs` source](https://github.com/NixOS/nixpkgs/blob/16af52ed24b47300d1bfc840b796c3cfe19ac945/pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/__init__.py), it's quite approachable. Related: https://github.com/NixOS/nixpkgs/issues/316651
